secant/Util/WalletStorage.swift

```diff
@@ -25,12 +34,105 @@ final class RecoveryPhraseStorage {
         case encoding
         case noDataFound
         case unknown(OSStatus)
-        case unsupportedVersion(Int)
-        case unsupportedLanguage(MnemonicLanguageType)
     }
     
-    // MARK: - Codable helpers
+    static func importWallet(
+        bip39 phrase: String,
+        birthday: BlockHeight?,
+        language: MnemonicLanguageType = .english,
+        hasUserPassedPhraseBackupTest: Bool = false
+    ) throws {
+        // Future-proof of the bundle to potentialy avoid migration. We enforce english mnemonic.
+        guard language == .english else {
+            throw WalletStorageError.unsupportedLanguage(language)
+        }
+
+        let wallet = StoredWallet(
+            birthday: birthday,
+            language: language,
+            seedPhrase: phrase,
+            hasUserPassedPhraseBackupTest: hasUserPassedPhraseBackupTest,
+            version: Constants.zcashKeychainVersion
+        )
+
+        do {
+            guard let data = try WalletStorage.encode(object: wallet) else {
+                throw KeychainError.encoding
+            }
+            
+            try setData(data, forKey: Constants.zcashStoredWallet)
+        } catch KeychainError.duplicate {
+            throw WalletStorageError.alreadyImported
+        } catch {
+            throw WalletStorageError.storageError(error)
+        }
+    }
     
+    static func exportWallet() throws -> StoredWallet {
+        guard let data = data(forKey: Constants.zcashStoredWallet) else {
+            throw WalletStorageError.uninitializedWallet
+        }
+        
+        guard let wallet = try WalletStorage.decode(json: data, as: StoredWallet.self) else {
+            throw WalletStorageError.uninitializedWallet
+        }
+        
+        guard wallet.version == Constants.zcashKeychainVersion else {
+            throw WalletStorageError.unsupportedVersion(wallet.version)
+        }
+        
+        return wallet
+    }
+    
+    static func areKeysPresent() throws -> Bool {
+        do {
+            _ = try exportWallet()
+        } catch {
+            // TODO: - report & log error.localizedDescription
```

Comment:
complete TODO or add Follow-up github ticket for it

Reply:
fixed